crates/core/src/blob/tree.rs

```diff
@@ -689,9 +690,11 @@ impl<P: Progress> TreeStreamerOnce<P> {
             let out_tx = out_tx.clone();
             let _join_handle = std::thread::spawn(move || {
                 for (path, id, count) in in_rx {
-                    out_tx
+                    if let Err(err) = out_tx
                         .send(Tree::from_backend(&be, &index, id).map(|tree| (path, tree, count)))
-                        .unwrap();
+                    {
+                        error!("Error sending tree {id} from backend: {err}");
```

Comment:
This makes rustic not panic anymore. But it will return `OK(())` from this method. I wonder if this might lead to rustic not performing a requested operation while only showing this error log but not returing any error...
Is there a way that we can also return `Err` here?

Reply:
Might by actually something to discuss in our general error topic: Handle errors in threads we just have spawned...

Reply:
I think we should use a channel to communicate with the main thread and send the errors there. In General. In this case, it's harder to judge. I think we need to panic!() here in one way, as I think a `send` not working out might be a bug in the program. So I agree, this might not be a good solution to just log.

Reply:
Panicking, but improved the message for the user by using `try_send`. We should investigate this more, this shouldn't panic if the channel `is connected` and `not full`. So either way, I improved this in a way, we are able to further diagnose the error in the future, should it happen again.